### PR TITLE
Make K8s integration asset centric

### PR DESCRIPTION
**Pull request by gizas**
- Enhancement

## Proposed commit message

WHAT: We change the structrure of the Kubernetes Integration ,trying to make it more resource centric. This means that the grouping of the configuration options for kubernetes collection (metrics or logss) is done based on the resource type that users want to monitor.  This done with such a way in order to help UX and become more user friendly. 

WHY:
See more info in https://github.com/elastic/integrations/issues/7936

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone dir
- cd packages/integrations/kubernetes
- elastic-package build
- elastic-package stack up -d -vvv --version=8.14.0-SNAPSHOT  
- Create an elastic-agent policy
- Add kubernetes integration. The version 1.58.0 should be available

### Testing upgrade procedure

I have also tested the upgrade procedure from an old k8s integration to this new structure

You can follow the steps:
1. Initially go and add an older version of the integration in your policy by using link
`https://localhost:5601/app/fleet/integrations/kubernetes-1.55.0/add-integration`
2. Install the elastic-agent
3. Go to Stack Management> Index Management > Data Streams
Choose Hidden datastreams

Screenshot 2024-03-13 at 5 24 05 PM

4. Then you can try upgrading to v1.58.0 
Choose the policy and click kubernetes option
Screenshot 2024-03-13 at 5 27 12 PM

5. Go to Settings and choose to upgrade to latest version
Screenshot 2024-03-13 at 5 27 31 PM

**6. Upgrade was successful and also the logs/metrics collection continued successfully**

## Related issues

- Closes #https://github.com/elastic/integrations/issues/7936

## Screenshots

How the full list of the datastream option looks like: 
fulllist

We group Cluster Resource Metrics
cluster_resource

Example of Pod Metrics

Pod_metrics

Small video for demonstration for UI teams:

https://github.com/elastic/integrations/assets/1685415/b733f00e-83a1-4e96-8bc6-60a8a333e286

### CODE REVIEW

**Comment by gizas** [comment]
@mlunadia and @roshan-elastic I ping you here in order to review the user interaction of the PR.

Please let me know which other people we should involve here in order to evaluate this effort and get valuable feedback for what we want to achieve

**Review by roshan-elastic — APPROVED** [review]
This makes so much more sense @gizas - nice job!

File: packages/kubernetes/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: 1.58.0
+  changes:
+    - description: Make Kubernetes integration settings resource centric
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/8835
```
Comment (tetianakravchenko) [inline packages/kubernetes/changelog.yml:6]:
```suggestion
      link: https://github.com/elastic/integrations/pull/9171
```

File: packages/kubernetes/data_stream/state_cronjob/manifest.yml [hunk 2]
```diff
@@ -69,5 +69,5 @@ streams:
         description: >
           Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the events are shipped. See [Processors](https://www.elastic.co/guide/en/beats/filebeat/current/filtering-and-enhancing-data.html) for details.
 
-    title: Kubernetes Cronjob metrics
+    title: Kube-state-metrics Cronjob metrics
     description: Collect Kubernetes Cronjob metrics from kube_state_metrics
```
Comment (tetianakravchenko) [inline packages/kubernetes/data_stream/state_cronjob/manifest.yml:73]:
```suggestion
    description: Collect Kubernetes Cronjob metrics from kube-state-metrics
```
just for follow the same format as everywhere else

File: packages/kubernetes/manifest.yml [hunk 3]
```diff
@@ -42,7 +59,7 @@ policy_templates:
               Condition to filter when to collect this input. See [Dynamic Input Configuration](https://www.elastic.co/guide/en/fleet/current/dynamic-input-configuration.html) for details.
               Note: This condition value will be appended to the individual metric condition values by an "and" clause if defined.
             type: text
-            multi: false
+            multi: true
```
Comment (tetianakravchenko) [inline packages/kubernetes/manifest.yml:62]: is it expected change in this PR?
Do we expect multiple conditions defined?

File: packages/kubernetes/manifest.yml [hunk 4]
```diff
@@ -117,65 +169,80 @@ policy_templates:
         title: Metricbeat Kubernetes Overview
         size: 1896x961
         type: image/png
-  - name: kube-proxy
-    title: Kube-proxy Metrics
-    description: Collect metrics from Kubernetes Proxy server with Elastic Agent.
+  - name: cluster-metrics
+    title: Cluster Resource  metrics
+    description: Collect other general Cluster metrics with Elastic Agent. It includes metrics from Kube-state-metrics and also system and volume metrics from Kubelet.
```
Comment (tetianakravchenko) [inline packages/kubernetes/manifest.yml:174]:
```suggestion
    description: Collect other general Cluster metrics with Elastic Agent. It includes metrics from Kube-state-metrics and also system metrics from Kubelet.
```
?

File: packages/kubernetes/manifest.yml [hunk 5]
```diff
@@ -55,34 +72,56 @@ policy_templates:
...
-    title: Kube-state-metrics
-    description: Collect container metrics from Kubernetes Kube-state-metrics with Elastic Agent.
+  - name: pod
+    title: Pod Metrics
+    description: Collect metrics for pod with Elastic Agent. It includes metrics from Kubernetes Kubelet API and from Kube-state-metrics.
     data_streams:
-      - state_container
+      - pod
+      - state_pod
       - state_cronjob
```
Comment (tetianakravchenko) [inline packages/kubernetes/manifest.yml:81]: why cronjob and job are defined in the Pod category?
isn't it the same as `state_deployment` or `state_daemonset` ?

File: packages/kubernetes/manifest.yml [hunk 6]
```diff
@@ -55,34 +72,56 @@ policy_templates:
...
+        description: Collecting Pod metrics from Kubelet and kube-state-metrics
+        input_group: metrics
+        vars:
+          - name: condition
+            title: Condition
+            description: |
+              Condition to filter when to collect this input. See [Dynamic Input Configuration](https://www.elastic.co/guide/en/fleet/current/dynamic-input-configuration.html) for details.
+              Note: This condition value will be appended to the individual metric condition values by an "and" clause if defined.
+            type: text
+            multi: true
```
Comment (tetianakravchenko) [inline packages/kubernetes/manifest.yml:95]: the same question as above

File: packages/kubernetes/manifest.yml (hunk 5 above)
Comment (gizas) [reply packages/kubernetes/manifest.yml:81]: Now that I see it again, maybe to move state_depoloyment and state_daeomonset under pod as well?

File: packages/kubernetes/manifest.yml (hunk 6 above)
Comment (gizas) [reply packages/kubernetes/manifest.yml]: reverted to false

File: packages/kubernetes/manifest.yml (hunk 3 above)
Comment (gizas) [reply packages/kubernetes/manifest.yml]: reverted to false

**Comment by gizas** [comment]
Closing as duplicate https://github.com/elastic/integrations/pull/9713
Sorry for the noise and multiple pings

**Pull request closed without merging**